### PR TITLE
For #17092: Dismiss toolbar menu when configuration change occurs

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by codrut-topliceanu** [comment]
👍 Works great (correctly dismisses the `BrowserMenu` popUp) for "more options" tab tray overflow and 3 dot menu. There are a few more places where `BrowserMenu` is used though (such as collections 3 dot menu and search engine radio button selection). Will this fix need to be implemented for those places too?

Secondly, while this patch fixes the [original issue](https://github.com/mozilla-mobile/fenix/issues/15344) I suppose a second patch is needed to remove the workaround for that issue in order to fix [the current problem](https://github.com/mozilla-mobile/fenix/issues/17092), right? 🤷‍♂️

**Comment by rocketsroger** [comment]
> 👍 Works great (correctly dismisses the `BrowserMenu` popUp) for "more options" tab tray overflow and 3 dot menu. There are a few more places where `BrowserMenu` is used though (such as collections 3 dot menu and search engine radio button selection). Will this fix need to be implemented for those places too?
> 
> Secondly, while this patch fixes the [original issue](https://github.com/mozilla-mobile/fenix/issues/15344) I suppose a second patch is needed to remove the workaround for that issue in order to fix [the current problem](https://github.com/mozilla-mobile/fenix/issues/17092), right? 🤷‍♂️

Tabtray already does dismiss it https://github.com/mozilla-mobile/fenix/blob/e188185b15e48622ed1c804777df6f48e78d365d/app/src/main/java/org/mozilla/fenix/tabtray/TabTrayDialogFragment.kt#L179.  Which is the other 3 dot menu?

**Comment by rocketsroger** [comment]
> > (such as collections 3 dot menu and search engine radio button selection). Will this fix need to be implemented for those places too?

**Comment by rocketsroger** [comment]
> But there may be other places where `BrowserMenu` is used that I haven't tested.
> Wish we could fix the problem at its root somehow. :(

**Comment by rocketsroger** [comment]
Looking into this more I think I'll fix the other two you raised in separate issues.  The reason here is the workaround for Android M and below broke both collection and search 3 dot menu. (Without rotation)  This is bad because just tapping on the overflow 3 dot menu is the normal use case and rotating is the less encountered use case.  

The workaround needs to be removed immediately and I will see if I can uplift it to beta to fix these issues in Android M or below.

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 1]
```diff
@@ -266,6 +272,12 @@ class HomeFragment : Fragment() {
         return view
     }
 
+    override fun onConfigurationChanged(newConfig: Configuration) {
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:275]: nit: Should we also add a test similar to the one that we have on `BrowserFragmentTest`?

**Review by Amejia481 — APPROVED** [review]
✅

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 2]
```diff
@@ -187,6 +191,8 @@ class HomeFragment : Fragment() {
         val activity = activity as HomeActivity
         val components = requireComponents
 
+        menuButton = view.findViewById<MenuButton>(R.id.menuButton)
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:194]: nit: I don't think you need to use findViewById. Synthetics should let you find the view in your `onConfigurationChanged` call itself.

**Review by jonalmeida — APPROVED** [review]
LGTM, just one question/nit. 🙂

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt (hunk 2 above)
Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:194]: Agreed.  That is much cleaner.  Thanks I'll update.

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt (hunk 1 above)
Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:275]: Sure, not sure why I didn't see that.  I'll add the test.

Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:275]: Sorry my bad I missed the test

**Pull request merged**